Adding a Beam version of CloudBigtableIO.

sduskis opened the pull request:
This implementation will replace the Dataflow v1 CloudBigtableIO.  It's a straight copy of the v1 CloudBigtableIO, with changes that relate to the differences between Dataflow v1 and Beam.  It also changes the package name for clarity.

(This is a duplicate of #1476 which was accidentally merged and then removed)

igorbernstein2 commented on bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/coders/CBTIOCoderRegistrar.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * A {@link CoderProviderRegistrar} for standard types used with {@link CloudBigtableIO}.
+ */
+@AutoService(CoderProviderRegistrar.class)
```
Does this need a service file in resources?

sduskis replied:
I successfully ran full read and write jobs before.  There is a META-INF/services/org.apache.beam.sdk.coders.CoderProviderRegistrar in the generated jar.  I guess there's some magic in auto-service artifact.

igorbernstein2 commented on bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java:
```diff
@@ -0,0 +1,1039 @@
+
+public class CloudBigtableIO {
+
+  enum CoderType {
```
This can be removed

igorbernstein2 commented on bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/AbstractCloudBigtableTableDoFn.java:
```diff
@@ -0,0 +1,108 @@
+  protected void logExceptions(Object context, RetriesExhaustedWithDetailsException exception) {
+    logRetriesExhaustedWithDetailsException(DOFN_LOG, String.valueOf(context), exception);
+  }
+}
```
This doesn't populate display data anymore. Is that ok?

igorbernstein2 commented on bigtable-dataflow-parent/bigtable-hbase-beam/README.md:
```diff
@@ -0,0 +1,80 @@
+# Cloud Bigtable + Beam
```
Please add a note that this is experimental

sduskis commented:
I did diffs for all main classes. 
- Configuration objects now match between beam and Dataflow
- I added displayData
- I removed CoderType

I also added notes that this is experimental.
Thanks for the review!
